### PR TITLE
extraction-5-create-3rd-transform-replace-jsx-attribute - CTR-20

### PR DESCRIPTION
**Pull request by kiprasmel**
how to review: https://github.com/pipedrive/CodeshiftCommunity/pull/9 (commit-by-commit is still best)

### CODE REVIEW

File: packages/reusable-transforms/src/replace-jsx-attribute/replace-jsx-attribute.spec.ts [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+import { buttonConfig } from "../replace-jsx-prop-and-its-value.config.cui-specific";
+
+/**
+ * TODO FIXME - instead of using specific configs (e.g. `buttonConfig`),
+ * generalize to test behavior not specific to existing configs.
+ */
+
+describe("@pipedrive/convention-ui-react@5.0.0 replaceJsxPropAndItsValue tranform", () => {
+    inlineTest<ConfigToModifyJSXAttributeAndItsValue>({
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/replace-jsx-attribute/replace-jsx-attribute.spec.ts:14]: Maybe can try separate files like in mui-codemod? https://github.com/mui-org/material-ui/tree/master/packages/mui-codemod/src/v5.0.0

File: community/@pipedrive__convention-ui-react/src/misc-utils/findJsxAttribute.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    if (attr.length > 1) {
+        console.error(
+            "\n\nCODEMODS ERROR - found multiple jsx attributes with the same name", //
+            attr.nodes().map(a => a.name.name),
+            "\n\n",
+            attr,
+            path,
+        );
+
+        throw new Error("CODEMODS - see the above error output.");
```
Comment (kiprasmel) [inline community/@pipedrive__convention-ui-react/src/misc-utils/findJsxAttribute.ts:15]: this prolly shouldn't error, i'll add a comment instead

Comment (kiprasmel) [reply community/@pipedrive__convention-ui-react/src/misc-utils/findJsxAttribute.ts]: done.

File: packages/reusable-transforms/src/replace-jsx-attribute/deletePropIfShouldBeDeleted.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        newValue === equalsIfShouldBeDeleted || //
+        newValue === equalsIfShouldBeDeletedAndCommented
+    ) {
+        const nameOfEnvVarThatDisablesComments = "CODEMODS_DO_NOT_ADD_INFO_FOR_DELETED_PROP" as const;
+
+        if (
+            newValue === equalsIfShouldBeDeletedAndCommented && //
+            !process.env[nameOfEnvVarThatDisablesComments]
+        ) {
+            const comment: string = joinWithDeep(
```
Comment (kiprasmel) [inline packages/reusable-transforms/src/replace-jsx-attribute/deletePropIfShouldBeDeleted.ts:66]: refactoring this atm

**Review by martikaljuve — CHANGES_REQUESTED** [review]
see if you can replace inline tests with separate files.

**Review by martikaljuve — APPROVED** [review]

**Pull request merged**